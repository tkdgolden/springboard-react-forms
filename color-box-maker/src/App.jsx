import { useState } from 'react'
import './App.css'
import BoxList from './BoxList.jsx'


const App = () => (
  <>
    <h1>App</h1>
    <BoxList />
  </>
);

export default App
