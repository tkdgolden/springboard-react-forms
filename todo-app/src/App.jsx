import { useState } from 'react'
import TodoList from './TodoList.jsx'
import './App.css'

const App = () => (
    <>
      <TodoList />
    </>
);

export default App
