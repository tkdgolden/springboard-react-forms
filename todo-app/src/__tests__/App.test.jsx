import { describe, expect, test } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom'
import App from "../App";
import { useState } from "react";

describe("App", () => {
    // smoketest
    test("renders", () => {
        render(<App />);
        const title = screen.getByText("List", { exact: false });
        expect(title.textContent).toBe("To Do List");
    });

    // snapshot
    test("hasn't changed", () => {
        const {asFragment} = render(<App />);
        expect(asFragment()).toMatchSnapshot();
    });
});