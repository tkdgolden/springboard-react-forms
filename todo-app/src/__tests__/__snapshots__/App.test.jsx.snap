// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > hasn't changed 1`] = `
<DocumentFragment>
  <h1>
    To Do List
  </h1>
  <div
    id="1"
  >
    <p>
      Tada
    </p>
    <button>
      X
    </button>
  </div>
  <form>
    <label
      for="task"
    >
      Task: 
    </label>
    <input
      id="task"
      name="task"
      type="text"
      value=""
    />
    <button>
      Enter
    </button>
  </form>
</DocumentFragment>
`;
